crates/nu-protocol/src/example.rs

```diff
@@ -1,8 +1,20 @@
 use crate::Value;
+use serde::{Deserialize, Serialize};
 
 #[derive(Debug)]
-pub struct Example {
-    pub example: &'static str,
-    pub description: &'static str,
+pub struct Example<'a> {
+    pub example: &'a str,
+    pub description: &'a str,
+    pub result: Option<Value>,
+}
+
+// PluginExample is somehow like struct `Example`, but it owned a String for `example`
+// and `description` fields, because these information is fetched from plugin, a third party
+// binary, nushell have no way to construct it directly.
+#[cfg(feature = "plugin")]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct PluginExample {
+    pub example: String,
+    pub description: String,
```

Comment:
Don't have the intuition if that works out with the lifetimes here, but could `PluginExample` and `Example` be unfied with a `Cow<str>` for serde? (Ripple effect would be that we would have to change all the command example code, so probably more academic)

Reply:
Like the idea!  But sadly when I try to apply `Cow<str>` in `PluginExample`, and fixed relative compile error, it finally ends with something like this error:
```
error[E0515]: cannot return value referencing local variable `encoding`
   --> crates/nu-plugin/src/plugin/mod.rs:174:18
    |
163 |     let response = encoding.decode_response(&mut buf_read)?;
    |                    --------------------------------------- `encoding` is borrowed here
...
174 |         Ok(_) => Ok(signatures),
    |                  ^^^^^^^^^^^^^^ returns a value referencing data owned by the current function
```

It makes some other struct like `PluginResponse` have a lifetime, and it outlives `encoding`...I don't think it can be fixed easily...

Reply:
Yeah then let's keep the current solution